### DISCUSSION TITLE
vscode: cannot access dbus secrets

### DISCUSSION THREAD

**Discussion by Dieterbe** [netblue30/firejail#5413] [Q&A]
When you install the GitHub Copilot plugin in visual studio code, it needs to authenticate via the github website. Once you do that, the GH website requests to open some local file/address which is opened by vscode, which completes authentication.
However, it also tries to write a secret.

With the stock profile, you get the error:

> Writing login information to the keychain failed with error 'GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.secrets was not provided by any .service files'.

I have tried adding this to the global profile, under all the ignore stuff, but this results in the same error message. However at this point it's possible i legitimately have nothing providing the secrets service.  I don't have time to research this further now, but perhaps someone spots an obvious PR opportunity.

```
dbus-user filter
dbus-user.talk org.freedesktop.secrets
```

**Comment by rusty-snake** [comment]
Sercret Service access shouldn't be blocked.

First make sure that you have an implementation (e.g. gnome-keyring or keepassxc) running (`busctl --user --acquired | grep org.freedesktop.secrets` should output something). And that firejail causes this, i.e. running w/o firejail works.

**Comment by ghost** [comment]
On Arch Linux [org.freedesktop.secrets](https://archlinux.org/packages/?q=org.freedesktop.secrets) is provided by `gnome-keyring`, `keepassxc` or `kwallet`. Do you have any of these installed?

**Comment by Dieterbe** [comment]
thanks guys..
I did not have any such tool installed.
I have now installed keepassxc. When i run it via firejail with the default profile, and i go to settings and try to enable the secret service integration, I get:
> Fdo Secret Service: Failed to register DBus service at org.freedesktop.secrets.
Another secret service is running (PID: Unknown, Executable: Unknown).
Please stop/remove it before re-enabling the Secret Service Integration.


and:
```
busctl --user --acquired | grep -E -i 'secret|keepass'                                                                                                                                                    ✘ 2 
org.keepassxc.KeePassXC.MainWindow            22751 0               dieter :1.45      user@1011.service 
```

However, when I run it outside of firejail, it seems to work (!) - so it seems that the default firejail profile blocks the dbus secrets integration.
```
~ ❯❯❯ busctl --user --acquired | grep -E -i 'secret|keepass'
org.freedesktop.secrets                       53448 keepassxc       dieter :1.49      user@1011.service -       -
org.keepassxc.KeePassXC.MainWindow            53448 keepassxc       dieter :1.49      user@1011.service -       -
```

**Reply by rusty-snake** [reply]
#3329
